app/controllers/admin/api/buyers_applications_controller.rb

```diff
@@ -158,6 +158,7 @@ def find
   ##~ op.parameters.add :name => "plan_id", :description => "ID of the new application plan.", :dataType => "int", :paramType => "query", :required => true, :threescale_name => "application_plan_ids"
   #
   def change_plan
+    application_plan = same_service_application_plans.find(params[:plan_id])
```

Comment:
Well that will return a 404.
Probably a 412 as suggested by @mikz  because if the plan exists then we should return an error instead.
I would do it in the `#change_plan!` instead of here

Reply:
I agree that 404 can be misleading and that anything enforced inside `#change_plan!` will be more effective.

Not sure about the 412 though. A 412 response usually means that the request does not match a precondition stated by the client itself in the request headers (in contraposition to something obscure defined by the server).

Reply:
[Now](https://github.com/3scale/porta/pull/568/commits/774f9676967f18dcda839e1c187fb46e029e08cc) it turns out it will be 422  :-)